Lib/idlelib/util.py

```diff
@@ -0,0 +1,5 @@
+def fix_scaling(root):
+    import tkinter.font
+    for name in tkinter.font.names(root):
+        font = tkinter.font.Font(root=root, name=name, exists=True)
+        font['size'] = abs(font['size'])
```

Comment:
On Windows, this is nonsensical, but I remember that when we switched the IDLE default font to a tk default font, a year ago, there was a bug on Linux that sizes were negative.  IDLE currently replaces negative sizes with 10.  Should the size be negated instead?

It seems to me that the bug should be fixed in tkinter.font.  Do you think that people are depending on the bug?

That aside, it is unclear how the function has any permanent effect after the function returns.  I will have to trust that it does.

I would prefer that the function be put in run.py rather than a tiny new module.  Pyshell already imports objects from run.py that are needed in both.  If doing so results in a circular import problem, and you want to get this in before the cuttoff times today, I can fiddle with the placement and imports later.

Reply:
> IDLE currently replaces negative sizes with 10. Should the size be negated instead?

Perhaps.

> It seems to me that the bug should be fixed in tkinter.font. Do you think that people are depending on the bug?

This is a delicate topic. Specifying negative size is a feature of Tk. It can be used for example for precise positioning texts together with bitmap images or other pixel graphics. And just negating it changes visible font size unless display has 72 DPI (very rarely in current days). On a display with 100 DPI a font with size 10 is 33% larger than a font with size -10.

I'll move `fix_scaling` to `run.py`.

Before merging I'm going to test changes on low-DPI display and on multi-display configuration. Would be nice to test it on OSX.